services/secret-service/server/src/main/resources/distribution/conf/secrets.json

```diff
@@ -31,5 +31,10 @@
     "type": "DROPBOX",
     "secretId": "dropbox-cred",
     "accessToken": ""
+  },
+  {
+    "type": "GDRIVE",
+    "secretId": "gdrive-cred",
+    "credentialsJson": ""
```

Comment:
1) From where is this JSON is retrieved. And what's included in it.

Reply:
So for Google Drive authentication, user needs to setup a service account which generates a secret.json file with fields below. User must copy paste the contents of the json file in the credentialsJson value. We are working on the documentation on how to setup a service account and how to use MFT for Gdrive transfers. We will update the documentation soon.
{
  "type": "service_account",
  "project_id": "",
  "private_key_id": "",
  "private_key": "",
  "client_email": "nsamft@nsamft.iam.gserviceaccount.com",
  "client_id": "",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": ""
}

Reply:
Have you tried accessing a file that is not public on web

Reply:
Hi Dinuka,
Yes we did try this. User needs to explicitly give permissions to a file/folder in Google drive for it to be accessed from outside system as it is a personal account and access of data should only be under the control of a user.

Reply:
How does the user give permission explicitly?

Reply:
Point 6 in this document explains about setting permission.
https://docs.google.com/document/d/1ql3megjxd_4_t9ID23se9Lb8NFpY83e175XHoKYpYdg/edit?usp=sharing

Reply:
+1

Reply:
I feel better if a temporary access token can be saved on the secret and use that to access the file. That's how it's  done in box too. So, later once a browser component for mft is available, the oauth flow can be added.